### PR TITLE
Add SN_FILTER_RESULTS  env settings.

### PR DESCRIPTION
**Pull request by HampusLundqvist**
I have to create many dynamic inventories in AWX/Tower, each with a different filter.
This feature would make it much more easy to automate creation of such inventories via e.g. the ansible.tower.tower_inventory_source module.
We would need only one .yml -file in the project and can configure filters on each inventory_source, via the "env" -settings or a special credential_type using env.

would it be possible to pull this in?

BR 
Hampus

### CODE REVIEW

**Comment by HampusLundqvist** [comment]
Hi
Is there anything more i have to do to get the workflow approved?
ping @amittell

**Review by amittell — APPROVED** [review]

File: plugins/inventory/now.py [hunk 1]
```diff
@@ -261,7 +265,9 @@ def parse(self, inventory, loader, path,
         selection = self.get_option('selection_order')
         fields = self.get_option('fields')
         table = self.get_option('table')
-        filter_results = self.get_option('filter_results')
+        filter_results = os.environ.get('SN_FILTER_RESULTS')
+        if not filter_results:
+            filter_results = self.get_option('filter_results')
```
Comment (willtome) [inline plugins/inventory/now.py]: `self.get_option` will look up from the environment variable and fall back to the option specified in the inventory file. The direct env lookup and if condition are redundant behavior.

**Review by willtome — CHANGES_REQUESTED** [review]
Thank you very much for the PR! I have 2 things to ask of you for approval. 
1) please see the comment on `self.get_option` usage and update accordingly
2) can you take a look at the [newer version](https://github.com/ansible-collections/servicenow.itsm/blob/main/plugins/inventory/now.py) of this plugin and see if the filtering method fits your use case? If it does not, can you raise an issue describing the functionality you need (including env var lookup of course)

Thanks!

File: plugins/inventory/now.py (hunk 1 above)
Comment (HampusLundqvist) [reply plugins/inventory/now.py]: > `self.get_option` will look up from the environment variable and fall back to the option specified in the inventory file. The direct env lookup and if condition are redundant behavior.

Hi
I tried setting the filter via filter_results environment variable and not specify a filter_results in now.yml
export filter_results="somefilter"
ansible-inventory -i now.yml  --graph -vvv
When I run ansible-inventory, no filter is applied and I can see that sysparm_query=  is empty when running ansible-inventory in verbose mode (i tried this with ansible-2.9.20).

edit:
Ah, key to that seems to be in the DOCUMENTATION section. cool, thanks!

**Comment by HampusLundqvist** [comment]
Hi!
Created an issue for this feature on servicenow.itsm https://github.com/ansible-collections/servicenow.itsm/issues/102

**Review by willtome — APPROVED** [review]

**Pull request merged**